subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaBasePlugin.java

```diff
@@ -336,6 +338,10 @@ private void configureBuildDependents(Project project) {
 
     private void configureTest(final Project project, final JavaPluginExtension javaPluginExtension) {
         project.getTasks().withType(Test.class).configureEach(test -> configureTestDefaults(test, project, javaPluginExtension));
+        project.getTasks().withType(Jar.class).configureEach(jar -> {
+            TaskCollection<Test> testTasks = project.getTasks().withType(Test.class);
+            testTasks.forEach(test -> test.shouldRunAfter(jar));
+        });
```

Comment:
``
```suggestion
        // Prefer executing Jar tasks first, so other projects that depend on jar can start earlier
        // Otherwise a single slow test task might reduce parallelism in other projects
        TaskCollection<Jar> jarTasks = project.getTasks().withType(Jar.class);
        project.getTasks().withType(Test.class).configureEach(test -> {
            test.shouldRunAfter(jarTasks);
        });
```

Reply:
Thanks for the suggestion!  I think this ought to live closer to where the jar task is registered though, as I think the conceptual goal is better expressed as "make the jar task run without waiting for tests" rather than "make the tests run later", but unfortunately the `shouldRunBefore()` method that could be called within the register methods configuration block without running afoul of the mutation guards doesn't exist.

Assuming the performance tests look good, I'll bring this up with the build tool team for consideration.